src/code/z_kankyo.c

```diff
@@ -900,9 +900,9 @@ void Environment_Update(GlobalContext* globalCtx, EnvironmentContext* envCtx, Li
         time = gSaveContext.dayTime;
 
         if (time > 0xC000 || time < 0x4555) {
-            gSaveContext.nightFlag = true;
+            gSaveContext.nightFlag = 1;
         } else {
-            gSaveContext.nightFlag = false;
+            gSaveContext.nightFlag = 0;
```

Comment:
Perhaps I missed this discussion in discord, but what's the motivation for no longer treating this as a boolean? It seems consistent in that it's either daytime or nighttime (which for this variable is defined between the times on either side of the range 0x4555 - 0xC000)

Reply:
its often compared to as == 1 as opposed to != 0, which leans more toward the side of it not being a boolean. perhaps we can make this more clear in the future by changing the name from night flag to something else

Reply:
relevant discussion from a while back https://github.com/zeldaret/oot/pull/545#discussion_r549807437

Reply:
I see, thanks. What about something like `nightState`? Although that's outside the scope of this PR, so this is fine as is

Reply:
tbh we can't know, maybe they used an enum, maybe they used booleans and used `== true` (not that uncommon in the code), most likely they used a mix of everything.

for consistency I just switched true/false to 1/0 since the 1/0 were prevailing

I wouldn't be opposed to renaming the field but I don't really care either, this isn't the most complicated concept to understand in the whole codebase :p